src/ManifoldsBase.jl

```diff
@@ -545,6 +545,42 @@ function manifold_function_not_implemented_message(M::Manifold, f, x...)
     return "$(f) not implemented on $(M)$(m)"
 end
 
+"""
+    mid_point(M::Manifold, p1, p2)
+
+Calculate the middle between the two point `p1` and `p2` from manifold `M`.
+By default uses [`log`](@ref), divides the vector by 2 and uses [`exp`](@ref).
+"""
+function mid_point(M::Manifold, p1, p2)
+    q = allocate(p1)
+    return mid_point!(M, q, p1, p2)
+end
+
+"""
+    mid_point!(M::Manifold, q, p1, p2)
+
+Calculate the middle between the two point `p1` and `p2` from manifold `M`.
+By default uses [`log`](@ref), divides the vector by 2 and uses [`exp!`](@ref).
+Saves the result in `q`.
+"""
+function mid_point!(M::Manifold, q, p1, p2)
+    X = allocate_result(M, log, p1, p2)
```

Comment:
Is this maybe overdoing the „avoid-allocation and pass a variable down“ approach? I already thought that on the pole and schild; do you think that's really necessary? If so, it can stay of course.

Reply:
I can't be 100% sure without some benchmarks but ultimately different things will dominate the calculation for different manifolds. I'll remove the 5-argument `mid_point!` for now then.

Reply:
If we notice a huge Time spent on allocations we can surely add that again. What do you think about the X-allocation in pole and schild? For them I thought the same, but they are also called far more often in the for loop, so I would be fine with keeping those.

Reply:
Which X-allocation? There are a few ( `X*=2` is an interesting one to solve) but I think it's OK to take care of allocations later.

Reply:
Let's keep them as they are, `schilds_ladder!` (and pole) are merely internal functions anyways (as opposed to `mid_point`).